arrow-udf-deno/README.md

```diff
@@ -0,0 +1,178 @@
+# Deno UDF for Apache Arrow
+
+[![Crate](https://img.shields.io/crates/v/arrow-udf-js.svg)](https://crates.io/crates/arrow-udf-deno)
+[![Docs](https://docs.rs/arrow-udf-js/badge.svg)](https://docs.rs/arrow-udf-deno)
+
+## Usage
+
+Add the following lines to your `Cargo.toml`:
+
+```toml
+[dependencies]
+arrow-udf-deno = "0.0.1"
+```
+
+Create a `Runtime` and define your JS functions in string form.
+Note that the function must be exported and its name must match the one you pass to `add_function`.
+
+```rust
+use arrow_udf_deno::{Runtime, CallMode};
+
+let mut runtime = Runtime::new();
+runtime
+    .add_function(
+        "gcd",
+        arrow_schema::DataType::Int32,
+        CallMode::ReturnNullOnNullInput,
+        r#"
+        export function gcd(a, b) {
+            while (b != 0) {
+                let t = b;
+                b = a % b;
+                a = t;
+            }
+            return a;
+        }
+        "#,
+    )
+    .await
+    .unwrap();
+```
+
+You can then call the JS function on a `RecordBatch`:
+
+```rust,ignore
+let input: RecordBatch = ...;
+let output: RecordBatch = runtime.call("gcd", input).await.unwrap();
+```
+
+If you print the input and output batch, it will be like this:
+
+```text
+ input     output
++----+----+-----+
+| a  | b  | gcd |
++----+----+-----+
+| 15 | 25 | 5   |
+|    | 1  |     |
++----+----+-----+
+```
+
+For set-returning functions (or so-called table functions), define the function as a generator:
+
+```rust
+use arrow_udf_deno::{Runtime, CallMode};
+
+let mut runtime = Runtime::new();
+runtime
+    .add_function(
+        "range",
+        arrow_schema::DataType::Int32,
+        CallMode::ReturnNullOnNullInput,
+        r#"
+        export function* range(n) {
+            for (let i = 0; i < n; i++) {
+                yield i;
+            }
+        }
+        "#,
+    )
+    .await
+    .unwrap();
+```
+
+You can then call the table function via `call_table_function`:
+
+```rust,ignore
+let chunk_size = 1024;
+let input: RecordBatch = ...;
+let mut outputs = runtime.call_table_function("range", input, chunk_size).await.unwrap();
+while let Some(result) = outputs.next().await {
+    let output: RecordBatch = result?;
+    // do something with the output
+}
+```
+
+If you print the output batch, it will be like this:
+
+```text
++-----+-------+
+| row | range |
++-----+-------+
+| 0   | 0     |
+| 2   | 0     |
+| 2   | 1     |
+| 2   | 2     |
++-----+-------+
+```
+
+The JS code will be run deno runtime.
+
+See the [example](examples/js.rs) for more details.
+
+## Using external libraries
+
+You can use external libraries in your JS code. By using a bundler as esbuild, you can bundle the external libraries into a single file and then pass the bundled file to `add_function`. See the [example](examples/sse) for more details.
+
+To run the example, first you need to compile the bundle using esbuild:
```

Comment:
@wangrunji0408 Do you have the mock sse server up and running `cargo run -p sse-server`?

Reply:
Yes, I still can't make it work...

Reply:
@wangrunji0408 It could be because it's trying to use the ipv6 to localhost I changed the port and the URL to use 127.0.01 instead of localhost, let me know if it fixes it

Reply:
Well, it works now. Thank you!